Add development environment setup guide

hillaryj opened the pull request:
# <!-- Use the title to describe PR changes in the imperative mood --> #

## 🗣 Description ##

<!-- Describe the "what" of your changes in detail. -->
<!-- To avoid scope creep, limit changes to a single goal. -->

Creates a guide with setup instructions for getting a CISA-provided mac laptop asset up and running. 
Updates some framing language in other documentation to make it more accessible to new users.

## 💭 Motivation and Context ##

<!-- Why is this change required? -->
<!-- What problem does this change solve? How did you solve it? -->
<!-- Link to any related issue(s) here to auto-close them on merge. -->
This guide previously existed in the google drive but was horrifically outdated. This update brings the guide in line with our open source policy and will help standardize future environment setup.

This also adds a link to the main README to include the excellent guide on writing git commit messages that the team uses.

For some of the other documentation guides, I added some extra language to make the topics more accessible to new users.

I also wanted to try out the new PR template 🎉 

## 🧪 Testing ##

<!-- How did you test your changes? How could someone else test this PR? -->

<!-- Include details of your testing environment, and the tests you ran to -->
<!-- see how your change affects other areas of the code, etc. -->
Documentation was linted and passed the pre-commit.
I tested the instructions as I wrote them, as well as got feedback from some of the interns.

## 🚥 Types of Changes ##

<!-- What types of changes does your code introduce? -->
<!-- Add any relevant types and remove any that don't apply. -->
<!-- Consider making a PR against the template in
<!-- https://github.com/cisagov/.github for new types. -->

* Documentation change

## ✅ Checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- Draft PRs should have one or more un-checked boxes. -->
<!-- If you're unsure about any of these, don't hesitate to ask. -->
<!-- We're here to help! -->

* [x] PR has an informative and human-readable title
* [x] Changes are limited to a single goal - _eschew scope creep!_
* [x] _All_ future TODOs are captured in code comments _and_ issues
* [x] Assignee and relevant labels have been added
* [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document
* [x] Code changes follow [cisagov code standards](https://github.com/cisagov/development-guide)
* [x] Repo and/or project documentation updated to reflect the changes in this PR
* [x] All new and existing tests pass

hillaryj committed:
Add mac dev setup instructions

Please note these are in-progress.

hillaryj committed:
Add mac dev setup instructions

Please note these are in-progress.

hillaryj committed:
Formatting updates

Updates formatting for markdown plus some minor fixes

hillaryj committed:
Formatting updates

Updates formatting for markdown plus some minor fixes

hillaryj committed:
Add link to new instructions

hillaryj committed:
Add link to new instructions

hillaryj committed:
Fix header weights

hillaryj committed:
Fix header weights

hillaryj committed:
Fix header weights

hillaryj committed:
Fix header weights

hillaryj committed:
Flesh out some sections plus placeholders

hillaryj committed:
Flesh out some sections plus placeholders

hillaryj committed:
Update instructions

Adds COOL setup instructions and some other minor updates

hillaryj committed:
Update instructions

Adds COOL setup instructions and some other minor updates

hillaryj committed:
Additional info and linting

hillaryj committed:
Additional info and linting

hillaryj committed:
Restore linting.

hillaryj committed:
Restore linting.

hillaryj committed:
Flesh out GPG-signing and fix header depth

Adds more/better instructions for setting up a signing key. Fixes header depths and linting per the cisagov Markdown linting rules.

hillaryj committed:
Flesh out GPG-signing and fix header depth

Adds more/better instructions for setting up a signing key. Fixes header depths and linting per the cisagov Markdown linting rules.

hillaryj committed:
Add dev env and commit message guides

Adds a link to the CISA-provided Mac setup guide. Adds a link to the excellent "How to Write a Git Commit Message" guidelines.

hillaryj committed:
Add dev env and commit message guides

Adds a link to the CISA-provided Mac setup guide. Adds a link to the excellent "How to Write a Git Commit Message" guidelines.

hillaryj committed:
Merge branch 'add-dev-env-setup' of https://github.com/cisagov/development-guide into add-dev-env-setup

hillaryj commented:
I rebased, since it was 3 months out of date, which added some extra commits.

hillaryj committed:
Remove useless line break changes

To make the PR more easily readable, revert some text rearrangement formatting changes.

jsf9k commented on README.md:
```diff
@@ -4,16 +4,18 @@
   * HCL
   * JavaScript
   * [Python](/languages/python)
+* [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/#seven-rules)
```
:+1:

jsf9k commented on dev_envs/mac-env-setup.md:
```diff
@@ -0,0 +1,258 @@
+# Setting up a Mac-based development environment #
+
+## Initial Account Setup (as Administrator) ##
```
The capitalization in these second-level (and lower) headings don't agree with what you did in line 1.

hillaryj replied:
Good catch! The Plain Language guide recommends using sentence case for headers - I'll update this.

hillaryj replied:
Resolved in 802d2dd2215abd0eda70209fe993f0b5d8227ad4

mcdonnnj commented on dev_envs/mac-env-setup.md:
````diff
@@ -0,0 +1,258 @@
+viewing the folders listed inside of the `/Users/` folder - in this example,
+the account name is **username**.
+
+```bash
````
```suggestion
```console
```

mcdonnnj commented on dev_envs/mac-env-setup.md:
````diff
@@ -0,0 +1,258 @@
+Then, you’ll add your **username** to the `/etc/sudoers` file using `visudo`
+like this:
+
+```sh
````
```suggestion
```console
```

mcdonnnj commented on dev_envs/mac-env-setup.md:
````diff
@@ -0,0 +1,258 @@
+Download XCode and Accept the License Agreement, then install the XCode
+Command Line Tools.
+
+```bash
````
```suggestion
```console
```

mcdonnnj commented on dev_envs/mac-env-setup.md:
````diff
@@ -0,0 +1,258 @@
+To set up a dev environment via the [CISA `laptop` script](https://github.com/cisagov/laptop/),
+execute the following in your terminal:
+
+```sh
````
```suggestion
```console
```

mcdonnnj commented on dev_envs/mac-env-setup.md:
````diff
@@ -0,0 +1,258 @@
+Open Terminal and install `brew` per the instructions from
+[Homebrew](https://brew.sh). You will probably have to do this with `sudo`.
+
+```sh
````
```suggestion
```console
```

mcdonnnj commented on dev_envs/mac-env-setup.md:
````diff
@@ -0,0 +1,258 @@
+These are automatically installed if you used the Automatic installation
+script above.
+
+```sh
````
```suggestion
```console
```

mcdonnnj commented on dev_envs/mac-env-setup.md:
````diff
@@ -0,0 +1,258 @@
+clone it as shown below. Then, stow all the packages from `.dotfiles` into
+your home directory per the [`.dotfiles` README](https://github.com/cisagov/.dotfiles).
+
+```bash
````
```suggestion
```console
```

mcdonnnj commented on dev_envs/mac-env-setup.md:
````diff
@@ -0,0 +1,258 @@
+To allow access to the saved OSX Keychain credential for command line
+usage, run:
+
+```bash
````
```suggestion
```console
```

mcdonnnj commented on dev_envs/mac-env-setup.md:
````diff
@@ -0,0 +1,258 @@
+
+Generate a key to sign your git commits and add it to your `~/.gitconfig`:
+
+```bash
````
```suggestion
```console
```

mcdonnnj commented on dev_envs/mac-env-setup.md:
````diff
@@ -0,0 +1,258 @@
+Follow the prompts for name and email address, using either your CISA
+or Trio email address. The output should look like:
+
+```bash
````
I would either remove formatting or use `console` for this kind of code block.

jsf9k replied:
I think `console` makes sense here.

mcdonnnj commented on dev_envs/mac-env-setup.md:
````diff
@@ -0,0 +1,258 @@
+
+Lastly, enable the global configuration option for commit signing:
+
+```bash
````
```suggestion
```console
```

mcdonnnj commented on dev_envs/mac-env-setup.md:
````diff
@@ -0,0 +1,258 @@
+
+Note: use `ed25519` instead of `rsa`
+
+```bash
````
```suggestion
```console
```

mcdonnnj commented on dev_envs/mac-env-setup.md:
````diff
@@ -0,0 +1,258 @@
+1. Use a passphrase you'll remember and/or save in your keychain
+1. Start the `ssh-agent` in the background and add your new credential:
+
+```bash
````
```suggestion
```console
```

mcdonnnj commented on dev_envs/mac-env-setup.md:
````diff
@@ -0,0 +1,258 @@
+
+Note: use `ed25519` instead of `rsa`
+
+```bash
````
```suggestion
```console
```

jsf9k replied:
I'd also put the comment _before_ the command here.

mcdonnnj commented on project_setup/README.md:
````diff
@@ -33,9 +33,16 @@ skeleton-python-library
+your new repository for publication.  The tool will print out each command it
+is issuing and its result.
+
+```bash
````
```suggestion
```console
```

jsf9k commented on dev_envs/mac-env-setup.md:
```diff
@@ -0,0 +1,258 @@
+
+### Account Permissions ###
+
+If you have a Standard user account, you’ll need to allow it access to run
```
I don't think "Standard" needs to be capitalized here.

hillaryj replied:
Resolved in 802d2dd2215abd0eda70209fe993f0b5d8227ad4

jsf9k commented on dev_envs/mac-env-setup.md:
```diff
@@ -0,0 +1,258 @@
+
+### iTerm2 (recommended) ###
+
+To set up iTerm2 - a MacOS terminal replacement with some nifty features, visit
```
The comma should probably be a closing dash, to match the opening dash.

hillaryj replied:
Good catch!

hillaryj replied:
Resolved in 802d2dd2215abd0eda70209fe993f0b5d8227ad4

jsf9k commented on dev_envs/mac-env-setup.md:
```diff
@@ -0,0 +1,258 @@
+
+Once installed, run the iTerm2 app and select the menu option
+`iTerm2 > Make iTerm2 Default Term`. NOTE: You may also need to set this
+setting as your Standard user account.
```
Standard :arrow_right: standard

hillaryj replied:
Resolved in 802d2dd2215abd0eda70209fe993f0b5d8227ad4

jsf9k commented on dev_envs/mac-env-setup.md:
```diff
@@ -0,0 +1,258 @@
+repository and then map your dotfiles to that directory. This makes it much
+easier to sync and set up.
+
+Using a `stow` based setup also allows for some nifty features like
```
`stow` based :arrow_right: `stow`-based

hillaryj replied:
Resolved in 802d2dd2215abd0eda70209fe993f0b5d8227ad4

jsf9k commented on dev_envs/mac-env-setup.md:
````diff
@@ -0,0 +1,258 @@
+
+```sh
+brew install stow
+
````
Why a blank line here?

hillaryj replied:
Resolved in 802d2dd2215abd0eda70209fe993f0b5d8227ad4

jsf9k commented on open-source-policy/practice.md:
```diff
@@ -1,33 +1,33 @@
-* CISA releases software into the [international public domain](#public-domain).
-* All team members should feel empowered to contribute back to outside open
-  source projects.
+* CISA releases software into the [international public domain](#public-domain)
```
Why remove the period?  This is a complete sentence.

hillaryj replied:
Aha, the word wrapping in my editor made the period on its own separate line. Fixed in 267f33866baf7df7d71137c503c07968d32d75e2

jsf9k commented on open-source-policy/practice.md:
```diff
@@ -1,33 +1,33 @@
+* Contribute back to outside open source projects whenever possible.
 * We [develop our software in the open](#working-in-public), while also
-  [protecting sensitive information](#protecting-sensitive-information).
+  [protecting sensitive information](#protecting-sensitive-information)
```
Why is the period removed?

hillaryj replied:
Resolved in 267f33866baf7df7d71137c503c07968d32d75e2

jsf9k commented on open-source-policy/practice.md:
```diff
@@ -1,33 +1,33 @@
+  [protecting sensitive information](#protecting-sensitive-information)
 * There are [narrow, documented exceptions](#exceptions) where we may delay or
-  withhold source code.
+  withhold source code
```
Period.

hillaryj replied:
Resolved in 267f33866baf7df7d71137c503c07968d32d75e2

jsf9k commented on open-source-policy/practice.md:
```diff
@@ -1,33 +1,33 @@
-capacity of a US government employee.
+There are potentially other cases where copyright is involved: where
+contractors produce the work, or where work was otherwise originally performed
+not in the capacity of a US government employee.
```
Is "not in the" :arrow_right: "outside the" less awkward here?

hillaryj replied:
Good change - updated in in 267f33866baf7df7d71137c503c07968d32d75e2

jsf9k commented on open-source-policy/practice.md:
```diff
@@ -39,43 +39,41 @@ government or non-government -- before writing custom tools. When existing
-    * Email cisagov-github-group@trio.dhs.gov for who the best contact is.
+1. See if there is an organizational CLA available
+1. Send the agreement to DHS's Office of General Counsel (OGC) for review
+    * Email cisagov-github-group@trio.dhs.gov for who the best contact is
```
I think "for who the best contact is" :arrow_right: "to determine the best contact" is less awkward.

hillaryj replied:
I like it! Updated in 267f33866baf7df7d71137c503c07968d32d75e2

jsf9k replied:
What I meant was
```
    * Email cisagov-github-group@trio.dhs.gov to determine the best contact.
```

jsf9k replied:
Changed in commit ad35141.

jsf9k commented on onboarding-checklist.md:
```diff
@@ -0,0 +1,145 @@
+
+**Buddy:** If you can’t complete any of the items on your checklist
+personally, _you are responsible for ensuring that someone with the correct_
+_access completes that item_.
```
I don't think the trailing `_` after "correct" or the leading one before "access" are necessary.

hillaryj replied:
Artifact of my editor - it didn't italicize across the line break without them. Resolved in 802d2dd2215abd0eda70209fe993f0b5d8227ad4

jsf9k replied:
I think we should keep the italics, I was just pointing out that they don't have to be stopped and started across a line break.

hillaryj replied:
Yep, the italics stayed! the extra underscores just made my editor display the italicization.

jsf9k replied:
No, the italics have been removed.

jsf9k replied:
Italics reanimated in commit a90236a.

jsf9k commented on onboarding-checklist.md:
```diff
@@ -0,0 +1,145 @@
+personally, _you are responsible for ensuring that someone with the correct_
+_access completes that item_.
+
+## New Person checklist ##
```
Inconsistent capitalization

hillaryj replied:
It should have been `NewPerson` but I ended up moving away from that naming entirely. Resolved in 802d2dd2215abd0eda70209fe993f0b5d8227ad4

jsf9k commented on onboarding-checklist.md:
```diff
@@ -0,0 +1,145 @@
+the Rules of Behavior, which is required before we can give you access to any
+systems, like Trio.
+- [ ] Go through the [FISMA-Ready Github guide](https://github.com/fisma-ready/github)
+to set up your Github account to be added to CISA
```
This is a complete sentence and could end with a period.

hillaryj replied:
Resolved in 802d2dd2215abd0eda70209fe993f0b5d8227ad4

jsf9k replied:
Doesn't appear to be changed, which is OK if you don't want to.

hillaryj replied:
Fixed in a848650149665216a068543dd1b50962d7ec2737

jsf9k commented on onboarding-checklist.md:
```diff
@@ -0,0 +1,145 @@
+June 2020)
+- [ ] [COOL Wiki](https://github.com/cisagov/cool-system/wiki/)
+- [ ] Set up your development environment on your CISA-provided Mac laptop
+(Link TBD)
```
I think you created the source for this link, so it could be added now.

hillaryj replied:
Resolved in 802d2dd2215abd0eda70209fe993f0b5d8227ad4

jsf9k commented on onboarding-checklist.md:
```diff
@@ -0,0 +1,145 @@
+they could conceivably complete in their first two/three weeks using their
+existing skills. Discuss the context with them, then make them the assignee.
+- [ ] Discuss suggestions for how the onboarding experience could have been
+improved and open as PRs on [the onboarding template]
```
I think `[the onboarding template]` is a broken link.  Also this is a complete sentence.

hillaryj replied:
Yup, the link tag didn't match the links at the bottom any more. Resolved in 802d2dd2215abd0eda70209fe993f0b5d8227ad4

jsf9k commented on onboarding-checklist.md:
```diff
@@ -0,0 +1,145 @@
+
+## Required items for all team members ##
+
+- [ ] Add them to recurring meetings that are relevant for them
+- [ ] Add them to the relevant [Github cisagov teams](https://github.com/orgs/cisagov/teams/)
```
These are complete sentences and could end with a period.

hillaryj replied:
Resolved in 802d2dd2215abd0eda70209fe993f0b5d8227ad4

jsf9k replied:
I don't see a change, but that's OK if that's your intent.

hillaryj replied:
Nope, just missed it. Thanks for the catch! Fixed in a848650149665216a068543dd1b50962d7ec2737

jsf9k reviewed (approved):
Great improvements here!

I have a host of small comments, most of which can be ignored if you disagree.

hillaryj committed:
Add FISMA-Ready Github guide

hillaryj committed:
Use plain language

Updates titles per the US plain language guide.
Simplifies some wording.
Fixes typos and grammar.

hillaryj committed:
Improve wording and grammar

jsf9k commented on README.md:
```diff
@@ -4,17 +4,23 @@
   * JavaScript
   * [Python](/languages/python)
 
+## Git, GitHub, and You ##
```
Do we want to make the capitalization of these headers agree with what we are doing in the other files in this PR?

hillaryj replied:
Yup, and I'm fixing the pre-existing headers too.

hillaryj replied:
Fixed in a848650149665216a068543dd1b50962d7ec2737

jsf9k commented on dev_envs/mac-env-setup.md:
````diff
@@ -0,0 +1,260 @@
+1. Save the generated key in the default location (e.g. `~/.ssh/id_ed25519`)
+1. Use a passphrase you'll remember and/or save in your keychain
+1. Start the `ssh-agent` in the background and add your new credential:
+
+```bash
+eval "$(ssh-agent -s)"
+ssh-add -K ~/.ssh/id_ed25519
+```
````
Some indenting will make this code block more clearly part of the last bullet:
```suggestion
    ```bash
    eval "$(ssh-agent -s)"
    ssh-add -K ~/.ssh/id_ed25519
    ```
```

hillaryj replied:
I wish I'd seen the "Commit suggestion" button - I've fixed this in a848650149665216a068543dd1b50962d7ec2737

jsf9k commented on onboarding-checklist.md:
```diff
@@ -0,0 +1,140 @@
+1. Create a new issue called "Onboard _Name_".
+1. View the raw source of this file.
+1. Copy the text after the line below to the issue.
+1. Remove any block of tasks that doesn't apply to the new team member's role
```
This is a complete sentence and could end in a period, like the other bullets in this block.

hillaryj replied:
Fixed in a848650149665216a068543dd1b50962d7ec2737

jsf9k commented on onboarding-checklist.md:
```diff
@@ -0,0 +1,140 @@
+  make things better.
+  - If you notice a problem (especially with things like documentation), you
+  are more than welcome to fix it!
+  - You can also file issues and pull requests on the [template onboarding checklist]
```
This is a complete sentence, which could end in a period.  This would make this bullet agree with the others in this block.

hillaryj replied:
Fixed in a848650149665216a068543dd1b50962d7ec2737

jsf9k replied:
You added the period, but removed `checklist]`, thus breaking the link.

hillaryj replied:
Huh, turns out it helps if you save your final changes before committing. 🙄

hillaryj replied:
8ad9ac2272f97c8062f2e23c0c6b43ddc21104de

hillaryj replied:
`final_FINAL_v2_reallyfinal`

jsf9k commented on onboarding-checklist.md:
```diff
@@ -0,0 +1,140 @@
+Once you're set up on Microsoft Teams, we recommend you join the following:
+
+- [ ] [CISA](https://teams.microsoft.com/l/team/19%3aa40fd338ea7a45de994a7c7ce9671fd4%40thread.skype/conversations?groupId=8c6c0b00-7244-46e1-bef9-76ce2a1906f5&tenantId=3ccde76c-946d-4a12-bb7a-fc9d0842354a)
+- [ ] [CISA - VM- Fusion Branch](https://teams.microsoft.com/l/team/19%3a2d4e9f3e321f4ef9a09f4ed94a32105c%40thread.skype/conversations?groupId=2992771f-df53-4097-87dd-75898a5830db&tenantId=3ccde76c-946d-4a12-bb7a-fc9d0842354a)
```
Is the hyphen after "VM" intentional, or should this be "CISA - VM Fusion Branch"?  Or "CISA - VM - Fusion Branch"?

hillaryj replied:
It used to be the MS Teams name for it, but has been updated to "Vulnerability Management" - I'll update.

hillaryj replied:
Fixed in a848650149665216a068543dd1b50962d7ec2737

jsf9k commented on onboarding-checklist.md:
```diff
@@ -0,0 +1,140 @@
+
+- [ ] Bookmark the [dev team project board] to track issues and work across
+multiple Github repositories
+- [ ] Bookmark [VM JIRA] to work with the rest of VM Fusion.
```
This bullet is a complete sentence but does not end in a period, whereas other bullets in this block that are complete sentences do.  We should be consistent, at least in the same bulleted list.

I'm for adding a period if the bullet is a complete sentence, but as long as we are globally consistent I'm OK with what the group decides.

hillaryj replied:
Resolved a848650149665216a068543dd1b50962d7ec2737

jsf9k commented on open-source-policy/practice.md:
```diff
@@ -96,118 +95,111 @@ public.
 There are more categories of controlled unclassified information to protect;
-those are just the kinds that we work with most often. [Here’s the complete
-list.](https://www.archives.gov/cui/registry)
+those are just the kinds that we work with most often. [Here’s the complete list.](https://www.archives.gov/cui/registry)
```
I'd move the period outside the link, but ignore if you disagree.

hillaryj replied:
I agree, and now the linter allows punctuation after a link without complaining, which is probably why Mark had it inside the link to start with.

hillaryj replied:
The page had also moved. I made the link more descriptive in line with the plain language guides and fixed the link in a848650149665216a068543dd1b50962d7ec2737.

jsf9k reviewed (approved):
Now it looks even better!

I have more comments, which are still minor things that you can ignore if you want.  Several comments are about being consistent as to whether we add periods at the end of bullets that are complete sentences.  I think we should be consistent in each bulleted list, and I'd like us to be consistent globally - but I will go with whatever the group decides.

Sorry for being nitpicky.  You added a lot of prose, which is naturally going to generate a lot of issues like this.  Feel free to do the same to me when the time comes.

hillaryj committed:
Update titles and bullet periods

hillaryj committed:
Update onboarding-checklist.md

jsf9k committed:
Reinstate italics

jsf9k committed:
Make sentence less awkward

dav3r commented on dev_envs/mac-env-setup.md:
```diff
@@ -0,0 +1,260 @@
+
+### Install XCode command line tools ###
+
+Download XCode and Accept the License Agreement, then install the XCode
```
`Accept the License Agreement` ➡️ `accept the license agreement`

hillaryj replied:
Resolved

dav3r commented on dev_envs/mac-env-setup.md:
```diff
@@ -0,0 +1,260 @@
+### Docker ###
+
+Download the latest stable [Docker for Mac](https://www.docker.com/docker-mac).
+Install it by double-clicking the downloaded `dmg` file and dragging to the
```
`... and dragging to the...` ➡️ `...and dragging the Docker application file to the...`

hillaryj replied:
Resolved

dav3r commented on dev_envs/mac-env-setup.md:
```diff
@@ -0,0 +1,260 @@
+To set up iTerm2 - a MacOS terminal replacement with some nifty features -
+visit the [iTerm2 download page](https://www.iterm2.com/downloads.html) to
+get the latest stable release. Install it by double-clicking the downloaded
+`dmg` file and dragging to the `Applications` folder.
```
```
...the downloaded `dmg` file and dragging to the `Applications` folder.
```
⬇️ 
```
...the downloaded `zip` file and dragging the extracted `iTerm` application file to the `Applications` folder.
```

hillaryj replied:
Resolved in a commit before 37987f9

dav3r commented on dev_envs/mac-env-setup.md:
```diff
@@ -0,0 +1,260 @@
+get the latest stable release. Install it by double-clicking the downloaded
+`dmg` file and dragging to the `Applications` folder.
+
+Once installed, run the iTerm2 app and select the menu option
```
Not sure if it's worth changing the verbiage here, but the app file itself is called `iTerm`, even though the app refers to itself as `iTerm2`.  It's slightly confusing either way.

hillaryj replied:
Good catch! I resolved this in one of the last five commits from 37987f9

dav3r commented on dev_envs/mac-env-setup.md:
```diff
@@ -0,0 +1,260 @@
+
+#### Prerequisites ####
+
+These are automatically installed if you used the Automatic installation
+script above.
```
```suggestion
These are automatically installed if you used the
[automatic package installation](#automatic-package-installation-recommended)
script above.
```

dav3r commented on dev_envs/mac-env-setup.md:
```diff
@@ -0,0 +1,260 @@
+1. [Configuring your browser for the COOL](https://github.com/cisagov/cool-system/wiki/Configuring-your-browser-for-the-COOL)
+
+After you have successfully completed the tasks above, follow
+[these instructions to access Guacamole](https://github.com/cisagov/cool-system/wiki/Accessing-an-assessment-environment-with-Guacamole).
```
We may want to clarify this slightly as:
```
[these instructions to access Guacamole](https://github.com/cisagov/cool-system/wiki/Accessing-an-assessment-environment-with-Guacamole).
Note: You will need to have been given access to one or more COOL environments
in order to access anything with Guacamole.
```

jsf9k replied:
I suggest "Note: " :arrow_right:  "Note that "

hillaryj replied:
Resolved in c1026d4

dav3r reviewed (approved):
Aside from the small changes I requested, this all looks fantastic - excellent work @hillaryj! 🎃

felddy reviewed (approved):
Wow.  Unifying all of this knowledge in one place is a huge win!  Thank you for curating it, and making it accessible!  💯 ⭐ 
I've learned a bunch of things just reviewing this!

mcdonnnj commented on README.md:
```diff
@@ -4,18 +4,24 @@
-## Tools and Services We Use ##
+## Git, GitHub, and you ##
+
+* [FISMA-Ready Github account setup](https://github.com/fisma-ready/github)
```
I think we should emphasize that for us you *do not* need to use your government e-mail address to create a GitHub account if you do not have one already. You should add it as an account e-mail however.

hillaryj replied:
We'll probably want to adopt and tailor the FISMA-ready guide to our own policies.

hillaryj replied:
In the onboarding checklist, I added wording around create-vs-add-email. 34f2ae9

mcdonnnj commented on dev_envs/mac-env-setup.md:
```diff
@@ -0,0 +1,260 @@
+Install it by double-clicking the downloaded `dmg` file and dragging to the
+`Applications` folder.
+
+### iTerm2 (recommended) ###
```
Before we tag this as recommended I want to be sure that a number of Dev Team members use iTerm2 regularly, if not primarily, so that it is not misleading to eventual new team members who may seek help with setup. Otherwise, I think it should be tagged as optional.

dav3r replied:
This is a good point.  I don't use iTerm2- not sure if anyone else does either.  I haven't felt the need to ditch Apple's Terminal app, but I might give iTerm2 a try one of these days.

jsf9k replied:
I use Arch, BTW.

hillaryj replied:
`Recommended` from the 18F guide and also because the author of the guide uses it and can support 😆

mcdonnnj commented on dev_envs/mac-env-setup.md:
```diff
@@ -0,0 +1,260 @@
+
+### iTerm2 (recommended) ###
+
+To set up iTerm2 - a MacOS terminal replacement with some nifty features -
```
```suggestion
To set up iTerm2 - a macOS terminal replacement with some nifty features -
```

mcdonnnj commented on dev_envs/mac-env-setup.md:
```diff
@@ -0,0 +1,260 @@
+
+### VPN configuration ###
+
+Once you're set up with the lab (CAL) VPN connection, we recommend you set up
+split-tunneling. Instructions and a generator script are located in the
+[vpn-config-gen repository](https://github.com/cisagov/vpn-config-gen).
```
There are still some additions needed to make the resulting config from `vpn-config-gen` a fully functional split tunnel.

jsf9k replied:
If so we should list those additions here, or point to a separate document where they are listed.

hillaryj replied:
This is digging into the recesses of time here, but I think that by the time I used it, there were no additional steps required.

mcdonnnj commented on dev_envs/mac-env-setup.md:
```diff
@@ -0,0 +1,260 @@
+
+#### Configure `git` ####
+
+To allow access to the saved OSX Keychain credential for command line
```
```suggestion
To allow access to the saved macOS Keychain credential for command line
```

mcdonnnj commented on dev_envs/mac-env-setup.md:
```diff
@@ -0,0 +1,260 @@
+#### Set up your favorite IDE ####
+
+- Install your favorite IDE and set up its preferences to your liking
+- For python development, please set up `black` for opinionated linting so
```
```suggestion
- For Python development, please set up `black` for opinionated linting so
```

mcdonnnj commented on dev_envs/mac-env-setup.md:
````diff
@@ -0,0 +1,260 @@
+Generate a key to sign your git commits and add it to your `~/.gitconfig`:
+
+```bash
+gpg --gen-key
````
I think we should encourage secure PGP key practices such as using subkeys. I use [this guide](https://blog.eleven-labs.com/en/openpgp-almost-perfect-key-pair-part-1/) as a basis for `gpg` key creation and management.

jsf9k replied:
Agreed.  I didn't do that when I initially set up my key, but it is the better way to do it.

hillaryj replied:
Captured this in Issue https://github.com/cisagov/development-guide/issues/36

mcdonnnj commented on dev_envs/mac-env-setup.md:
````diff
@@ -0,0 +1,260 @@
+gpg --gen-key
+```
+
+Follow the prompts for name and email address, using either your CISA
````
If you are using a GitHub noreply address, you will need to add it to your key to get that sweet `Verified` mark on your commits.

hillaryj replied:
The FISMA-ready GitHub guide talks about a process to set up using your gov email address - this is probably worth discussing and coming up with a standard.

hillaryj replied:
Captured in Issue https://github.com/cisagov/development-guide/issues/36

mcdonnnj commented on dev_envs/mac-env-setup.md:
````diff
@@ -0,0 +1,260 @@
+1. Use a passphrase you'll remember and/or save in your keychain
+1. Start the `ssh-agent` in the background and add your new credential:
+
+    ```bash
````
```suggestion
    ```console
```

mcdonnnj commented on onboarding-checklist.md:
```diff
@@ -0,0 +1,140 @@
+1. View the raw source of this file.
+1. Copy the text after the line below to the issue.
+1. Remove any block of tasks that doesn't apply to the new team member's role.
+1. Submit the issue.
```
Where should the issue be created? In that repo, wouldn't it be better to add the text below as an Issue template as we have moved to for other repeat tasks?

hillaryj replied:
Moved the checklist into an issue template in 34f2ae9

mcdonnnj commented on onboarding-checklist.md:
```diff
@@ -0,0 +1,140 @@
+
+- [ ] Take notes on anything that is confusing or frustrating during your
+onboarding process.
+  - Please share this information with your buddy and/or the team so we can
```
```suggestion
  - Please share this information with your buddy and the team so we can
```
I think it's good to encourage a collaborative atmosphere.

mcdonnnj commented on onboarding-checklist.md:
```diff
@@ -0,0 +1,140 @@
+the Rules of Behavior, which is required before we can give you access to any
+systems, like Trio.
+- [ ] Go through the [FISMA-Ready Github guide](https://github.com/fisma-ready/github)
+to set up your Github account to be added to CISA.
```
```suggestion
to set up your GitHub account to be added to CISA.
```

mcdonnnj commented on onboarding-checklist.md:
```diff
@@ -0,0 +1,140 @@
+multiple Github repositories.
+- [ ] Bookmark [VM JIRA] to work with the rest of VM Fusion.
+- [ ] In addition to your CISA email address `first.last@cisa.dhs.gov`, you'll
+also get an account on Trio, which is a Google-based account used for
```
```suggestion
also get an account on Trio, which is a Google Workspace account used for
```

mcdonnnj commented on onboarding-checklist.md:
```diff
@@ -0,0 +1,140 @@
+  [chat.google.com](https://chat.google.com) and join applicable rooms.
+- [ ] Make sure you have been added to several [dev team Google Groups] so you
+can participate in team-wide internal communication:
+  - [ ] [ALL Trio users@beta.dhs.gov](https://groups.google.com/a/beta.dhs.gov/d/forum/all)
```
```suggestion
  - [ ] [ALL Trio users @ beta.dhs.gov](https://groups.google.com/a/beta.dhs.gov/d/forum/all)
```
I thought it was an e-mail address at first glance.

jsf9k replied:
Perhaps "ALL Trio users at `beta.dhs.gov`" makes sense for these links?  Using the "@" seems superfluous here.

hillaryj replied:
I'd copied these from the google groups page, will simplify all of them for clarity.

mcdonnnj commented on onboarding-checklist.md:
```diff
@@ -0,0 +1,140 @@
+  - [ ] [CISA COOL](https://groups.google.com/a/trio.dhs.gov/forum/#!forum/cisa-cool-group)
+  - [ ] [cisagov-github](https://groups.google.com/a/trio.dhs.gov/forum/#!forum/cisagov-github-group)
+  - [ ] [NCATS@beta.dhs.gov](https://groups.google.com/a/beta.dhs.gov/d/forum/ncats)
+  - [ ] [NCATS AWS Support@beta.dhs.gov](https://groups.google.com/a/beta.dhs.gov/d/forum/ncats-aws-support)
```
```suggestion
  - [ ] [NCATS AWS Support @ beta.dhs.gov](https://groups.google.com/a/beta.dhs.gov/d/forum/ncats-aws-support)
```
Same thing about an e-mail address.

mcdonnnj commented on onboarding-checklist.md:
```diff
@@ -0,0 +1,140 @@
+  - [ ] [cisagov-github](https://groups.google.com/a/trio.dhs.gov/forum/#!forum/cisagov-github-group)
+  - [ ] [NCATS@beta.dhs.gov](https://groups.google.com/a/beta.dhs.gov/d/forum/ncats)
+  - [ ] [NCATS AWS Support@beta.dhs.gov](https://groups.google.com/a/beta.dhs.gov/d/forum/ncats-aws-support)
+  - [ ] [NCATS Infrastructure Development@beta.dhs.gov](https://groups.google.com/a/beta.dhs.gov/d/forum/ncats-dev)
```
```suggestion
  - [ ] [NCATS Infrastructure Development @ beta.dhs.gov](https://groups.google.com/a/beta.dhs.gov/d/forum/ncats-dev)
```
One more!

hillaryj replied:
Resolved by changes in 2768312

mcdonnnj commented on onboarding-checklist.md:
```diff
@@ -0,0 +1,140 @@
+- [ ] Identify a straightforward, well-defined issue that involves
+their skills domain, schedule a meeting with the owner for an introduction (if
+it's not you), and set up pairing sessions several times a week to start.
+- [ ] Identify a straightforward, well-defined first story, ideally something
```
This language assumes familiarity with commercial Agile process development practices. Although common, this is not guaranteed, and we do not (currently) make extensive use of it in our development life-cycle.

hillaryj replied:
Addressed in 34f2ae9

mcdonnnj commented on onboarding-checklist.md:
```diff
@@ -0,0 +1,140 @@
+  - [ ] [CISA AWS Accounts](https://docs.google.com/spreadsheets/d/1It0FIlG7ZxTwrRV-zVYUAMw18L6ZstQiAYT7vGYf5VE/edit#gid=2069139012)
+  and provide one-time credentials.
+  - Note: AWS user names should be identical across accounts so that
+  permissions can be correctly managed by terraform
```
```suggestion
  permissions can be correctly managed by Terraform.
```

mcdonnnj commented on open-source-policy/README.md:
```diff
@@ -1,23 +1,20 @@
+In general, we work in the open by default.
 
-**[Read CISA's open source policy.](policy.md)**
+**[Read CISA's open source policy](policy.md)** for the official
+[Open Source Policy](policy.md) of [CISA](https://cisa.gov/) and more details
```
I think there only needs to be one link to `policy.md` here.

hillaryj replied:
Resolved in 37987f9

mcdonnnj commented on dev_envs/mac-env-setup.md:
````diff
@@ -0,0 +1,260 @@
+script above.
+
+```sh
+brew install stow
+brew install s3cmd
+brew install figlet
+brew install pinentry-mac
````
I'm not sure if this was intentional for clarity, but I believe these can be rolled into one command.
```suggestion
brew install figlet pinentry-mac s3cmd stow
```

mcdonnnj reviewed (changes requested):
I ran a first pass over everything (whew, it's a lot!), and this is some really great process documentation! ✨

I had some feedback, a lot of which (probably most) are changing `bash` and `sh` code block formatting to `console`. I'm of the opinion that the `bash` and `sh` formatting should be used for scripts, while the `console` formatting should be used for command line usage examples. This is what has happened in other `cisagov` repos, but I don't think anyone has touched up the markdown in this project to follow suite.

jsf9k commented:
> I'm of the opinion that the `bash` and `sh` formatting should be used for scripts, while the `console` formatting should be used for command line usage examples.

I'm a little more flexible about this.  If I particularly want to highlight some shell fu in a command line I may use `bash` or `sh` formatting instead of `console`.  `console` is _most_ effective at highlighting a command _immediately followed by the resulting output_; in a case like that I would definitely prefer `console`.

We need to codify these heretofore unwritten rules in a document once we agree on them.

hillaryj committed:
Update dev_envs/mac-env-setup.md

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

hillaryj committed:
Apply suggestions from code review

Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>

hillaryj committed:
Simplify group names for clarity

hillaryj committed:
Update mac-env-setup.md

Integrate suggested changes.

hillaryj committed:
Apply suggestions from code review

Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>

hillaryj committed:
Update README.md

Remove some of the redundancy from the Department of Redundancy Department.

hillaryj committed:
Update onboarding-checklist.md

Moves checklist to an issue template and changes wording to reflect this.
Breaks out GitHub section in pre-reqs.

hillaryj committed:
Update onboarding-checklist.md

Separates Trio setup into its own section.
Simplifies language around first issue.
Separates and adds some buddy tasks.

hillaryj commented:
I've captured some of the style discussion here in a proto style guide, which I'll continue adding to and add in a different PR.

mcdonnnj commented on .github/ISSUE_TEMPLATE/onboarding-checklist.md:
```diff
@@ -0,0 +1,146 @@
+#### Mandatory Training ####
+
+- [ ] Complete [PALMS Cybersecurity Training](https://etms.hq.dhs.gov),
+including Mandatory Cyber Security and Privacy Training, including accepting
```
```suggestion
including Mandatory Cyber Security and Privacy Training, and accepting
```

mcdonnnj commented on .github/ISSUE_TEMPLATE/onboarding-checklist.md:
```diff
@@ -0,0 +1,146 @@
+### Getting to know the VM Fusion Dev Team ###
+
+- [ ] Bookmark the [dev team project board] to track issues and work across
+multiple Github repositories.
```
```suggestion
multiple GitHub repositories.
```

mcdonnnj commented on .github/ISSUE_TEMPLATE/onboarding-checklist.md:
```diff
@@ -0,0 +1,146 @@
+- [ ] If you don't already have a GitHub account,
+[create a GitHub account with your government email address](https://github.com/join).
+- [ ] If you already have a GitHub account, [add your government email address](https://docs.github.com/en/free-pro-team@latest/github/setting-up-and-managing-your-github-user-account/adding-an-email-address-to-your-github-account).
+- [ ] Go through the [FISMA-Ready Github guide](https://github.com/fisma-ready/github)
```
```suggestion
- [ ] Go through the [FISMA-Ready GitHub guide](https://github.com/fisma-ready/github)
```

mcdonnnj commented on .github/ISSUE_TEMPLATE/onboarding-checklist.md:
```diff
@@ -0,0 +1,146 @@
+## Required items for all team members ##
+
+- [ ] Invite them to relevant recurring meetings.
+- [ ] Invite them to relevant [Github cisagov teams](https://github.com/orgs/cisagov/teams/).
```
```suggestion
- [ ] Invite them to relevant [GitHub cisagov teams](https://github.com/orgs/cisagov/teams/).
```

mcdonnnj commented on .github/ISSUE_TEMPLATE/onboarding-checklist.md:
```diff
@@ -0,0 +1,146 @@
+Useful links:
+
+- [ ] [NCATS Data Dictionary](https://github.com/cisagov/ncats-data-dictionary)
+repo for information about what's stored in the MongoDB database
```
```suggestion
repo for information about what's stored in the CyHy environment MongoDB databases
```

mcdonnnj commented on README.md:
```diff
@@ -4,18 +4,24 @@
-## Tools and Services We Use ##
+## Git, GitHub, and you ##
+
+* [FISMA-Ready Github account setup](https://github.com/fisma-ready/github)
```
```suggestion
* [FISMA-Ready GitHub account setup](https://github.com/fisma-ready/github)
```

mcdonnnj commented on dev_envs/mac-env-setup.md:
````diff
@@ -0,0 +1,262 @@
+execute the following in your terminal:
+
+```console
+bash <(curl --silent https://raw.githubusercontent.com/cisagov/laptop/master/laptop)
````
Does this need to be `/bin/bash` now as well? The manual install instructions below use `/bin/bash` instead.

jsf9k replied:
I'd also suggest using `bash` here vice `console`, given the bash-fu in the line.

hillaryj replied:
Good catch! Resolved in https://github.com/cisagov/development-guide/pull/35/commits/d7d2e3fb638a5c6e5480b00c9e427030ef000baf

mcdonnnj commented on dev_envs/mac-env-setup.md:
```diff
@@ -0,0 +1,262 @@
+
+#### Generate `ssh` key to use git on the command line ####
+
+Reference: [Github: Generating a new SSH key](https://docs.github.com/en/github/authenticating-to-github/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent#adding-your-ssh-key-to-the-ssh-agent)
```
```suggestion
Reference: [GitHub: Generating a new SSH key](https://docs.github.com/en/github/authenticating-to-github/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent#adding-your-ssh-key-to-the-ssh-agent)
```

mcdonnnj commented on dev_envs/mac-env-setup.md:
```diff
@@ -0,0 +1,262 @@
+
+#### Add ssh key to GitHub ####
+
+Reference: [Github: Adding a new SSH key to your GitHub account](https://docs.github.com/en/github/authenticating-to-github/adding-a-new-ssh-key-to-your-github-account)
```
```suggestion
Reference: [GitHub: Adding a new SSH key to your GitHub account](https://docs.github.com/en/github/authenticating-to-github/adding-a-new-ssh-key-to-your-github-account)
```

mcdonnnj commented on dev_envs/mac-env-setup.md:
````diff
@@ -0,0 +1,262 @@
+$ pbcopy < ~/.ssh/id_ed25519.pub
+```
+
+1. Load your [Github Settings page for adding an SSH key](https://github.com/settings/ssh/new)
````
```suggestion
1. Load your [GitHub Settings page for adding an SSH key](https://github.com/settings/ssh/new)
```

mcdonnnj commented on open-source-policy/practice.md:
```diff
@@ -1,33 +1,33 @@
-CISA intentionally waives this copyright abroad using [Creative Commons Zero
+States government are not copyrightable in the US, and so are public domain.
+But by default, US government works **are** copyrightable internationally, and
+so CISA intentionally waives this copyright abroad using [Creative Commons Zero
 (CC0) 1.0](https://creativecommons.org/publicdomain/zero/1.0/).
```
```suggestion
so CISA intentionally waives this copyright abroad using
[Creative Commons Zero (CC0) 1.0](https://creativecommons.org/publicdomain/zero/1.0/).
```

mcdonnnj commented on open-source-policy/practice.md:
```diff
@@ -96,118 +94,112 @@ public.
+must agree to release their contributions into the international public
+domain. Projects can inform contributors of this agreement by copying the
+[`CONTRIBUTING.md`](CONTRIBUTING.md) file from this repo into new project
+repos, and copying the ["Public domain" section of this repo's README](README.md#public-domain)
```
```suggestion
repos, and copying the
["Public domain" section of this repo's README](README.md#public-domain)
```

mcdonnnj commented on open-source-policy/practice.md:
```diff
@@ -96,118 +94,112 @@ public.
-public](https://18f.gsa.gov/2014/07/31/working-in-public-from-day-1/). It
-creates a healthier working environment, a more collaborative process, and just
-better software.
+CISA believes in [working in public](https://18f.gsa.gov/2014/07/31/working-in-public-from-day-1/)
```
```suggestion
CISA believes in
[working in public](https://18f.gsa.gov/2014/07/31/working-in-public-from-day-1/)
```

mcdonnnj reviewed (approved):
A bunch of little suggestions and a question. Once again, strong work putting this all together ✨

hillaryj commented on dev_envs/mac-env-setup.md:
````diff
@@ -0,0 +1,262 @@
+To set up a dev environment via the [CISA `laptop` script](https://github.com/cisagov/laptop/),
+execute the following in your terminal:
+
+```console
````
```suggestion
```bash
```

hillaryj committed:
Apply suggestions from code review

Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>

hillaryj committed:
Apply suggestions from code review

Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>
